tests/dragonfly/http_conf_test.py

```diff
@@ -2,46 +2,62 @@
 
 
 async def test_password(df_factory):
-    # Needs a private key and certificate.
     with df_factory.create(port=1112, requirepass="XXX") as server:
         async with aiohttp.ClientSession() as session:
             resp = await session.get(f"http://localhost:{server.port}/")
             assert resp.status == 401
         async with aiohttp.ClientSession(
-            auth=aiohttp.BasicAuth("user", "wrongpassword")
+            auth=aiohttp.BasicAuth("default", "wrongpassword")
         ) as session:
             resp = await session.get(f"http://localhost:{server.port}/")
             assert resp.status == 401
-        async with aiohttp.ClientSession(auth=aiohttp.BasicAuth("user", "XXX")) as session:
+        async with aiohttp.ClientSession(auth=aiohttp.BasicAuth("default", "XXX")) as session:
             resp = await session.get(f"http://localhost:{server.port}/")
             assert resp.status == 200
 
 
+async def test_skip_metrics(df_factory):
+    with df_factory.create(port=1112, admin_port=1113, requirepass="XXX") as server:
+        async with aiohttp.ClientSession(auth=aiohttp.BasicAuth("whoops", "whoops")) as session:
+            resp = await session.get(f"http://localhost:{server.port}/metrics")
+            assert resp.status == 200
+        async with aiohttp.ClientSession(auth=aiohttp.BasicAuth("whoops", "whoops")) as session:
+            resp = await session.get(f"http://localhost:{server.admin_port}/metrics")
+            assert resp.status == 200
+
+
 async def test_no_password_on_admin(df_factory):
-    # Needs a private key and certificate.
     with df_factory.create(
         port=1112,
         admin_port=1113,
         requirepass="XXX",
-        noprimary_port_http_enabled=None,
+        primary_port_http_enabled=True,
         admin_nopass=None,
     ) as server:
-        async with aiohttp.ClientSession(auth=aiohttp.BasicAuth("user", "XXX")) as session:
+        async with aiohttp.ClientSession(auth=aiohttp.BasicAuth("default", "XXX")) as session:
             resp = await session.get(f"http://localhost:{server.admin_port}/")
             assert resp.status == 200
+        async with aiohttp.ClientSession(auth=aiohttp.BasicAuth("default")) as session:
```

Comment:
This was an existing bug, I fixed it. Basically, we required `auth` when df started with `primary_port_enabled` and `admin_nopass=True`.

Moreover, the previous test made use of non existing flag `noprimary_port_http_enabled=None`

Reply:
Re/ `noprimary_port_http_enabled` - that's a Google flags feature. You could pass any boolean flag with `no` prefix and it sets it to false